### PR TITLE
Tensorflowcudnn

### PR DESCRIPTION
**Pull request by teh**
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -1027,11 +1027,11 @@ in modules // {
   } else null;
 
   funcsigs = buildPythonPackage rec {
-    name = "funcsigs-0.4";
+    name = "funcsigs-1.0.2";
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: this might break a lot of packages

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -14066,11 +14066,11 @@ in modules // {
   };
 
   mock = buildPythonPackage (rec {
-    name = "mock-1.3.0";
+    name = "mock-2.0.0";
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: and so might this.

**Review by FRidh — COMMENTED** [review]
Could you run `nox-review pr 19701` on this and report the result?

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -30090,6 +30090,41 @@ in modules // {
...
+    version = "0.11.0rc0";
+    format = "wheel";
+
+    src = pkgs.fetchurl {
+      url = "https://storage.googleapis.com/tensorflow/linux/gpu/tensorflow-${version}-cp27-none-linux_x86_64.whl";
+      sha256 = "1r8zlz95sw7bnjzg5zdbpa9dj8wmp8cvvgyl9sv3amsscagnnfj5";
+    };
+
+    buildInputs = with self; [ pkgs.swig ];
+    propagatedBuildInputs = with self; [ numpy six protobuf3_0 pkgs.cudatoolkit75 pkgs.cudnn5_cudatoolkit75 pkgs.gcc49 self.mock ];
```
Comment (teh) [inline pkgs/top-level/python-packages.nix]: Yea, not sure why but fails at runtime without.

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (teh) [inline pkgs/top-level/python-packages.nix]: Do you know how to test this? I run

```
nox-review pr 19701
```

but it fails for all kinds of unrelated reasons. E.g. 

```
error: build of ‘/nix/store/z7gskp44ag26ydi2ydyd29ng2vydkhz9-android-sdk-25.2.2.drv’ failed
```

Comment (teh) [inline pkgs/top-level/python-packages.nix]: been running 

```
_NIX_OPTIONS="max-jobs=6" nox-review --keep-going pr 19701
```

for several hours now, seems to download all versions of eclipse, total 100G on disk so far. Feels like I should run a hydra cluster! :)

Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: yeah okay, you don't have to keep it running if it is enormous. It's kind of what I expected with an upgrade of mock and functools.

Comment (teh) [inline pkgs/top-level/python-packages.nix]: OK, build finished, and I restarted a few times to make sure I got a stable point. I now get these errors:

https://gist.github.com/teh/d1d4fe3bcae794581abea0d6adbaf329

But bizarrely when I pick out individual ones they build fine. Do you have any way of approaching this? I could of course add `mock-2_0_0` but at some point we probably have to make the jump, no?

**Comment by FRidh** [comment]
> OK, build finished, and I restarted a few times to make sure I got a stable point. I now get these errors:

Thanks for checking it. I built all Python packages [as well](https://headcounter.org/hydra/eval/337813) and there weren't actually any new packages failing.

> at some point we probably have to make the jump, no?

Definitely. Let's just do it.

**Comment by teh** [comment]
Thanks! Ping this thread if anything needs attention, will fix ASAP.

**Pull request merged**